arrow-core/src/main/kotlin/arrow/core/extensions/setk.kt

```diff
@@ -109,3 +141,108 @@ interface SetKEqK : EqK<ForSetK> {
       }
     }
 }
+
+/**
+ * [Semigroup] instance definition for [SetK]
+ */
+fun <A> Semigroup.Companion.setK(): Semigroup<SetK<A>> = Semigroup<SetK<A>> { b ->
+  this.fix().setCombineK(b)
+}
+
+/**
+ * [Monoid] instance definition for [SetK]
+ */
+fun <A> Monoid.Companion.setK(): Monoid<SetK<A>> = object : Monoid<SetK<A>>, SetKSemigroup<A> {
+  override fun empty(): SetK<A> =
+    emptySet<A>().k()
+}
+
+/**
+ * [Eq] instance definition for [SetK]
+ */
+fun <A> Eq.Companion.setK(EQ: () -> Eq<A>): Eq<SetK<A>> = object : Eq<SetK<A>> {
+  override fun SetK<A>.eqv(b: SetK<A>): Boolean =
+    if (size == b.size) map { aa ->
+      b.find { bb -> EQ().run { aa.eqv(bb) } } != null
+    }.fold(true) { acc, bool ->
+      acc && bool
+    }
+    else false
+}
+
+/**
+ * [Show] instance definition for [SetK]
+ */
+fun <A> Show.Companion.setK(SA: () -> Show<A>): Show<SetK<A>> = object : Show<SetK<A>> {
+  override fun SetK<A>.show(): String =
+    show(SA())
+}
+
+/**
+ * [Foldable] instance definition for [SetK]
+ */
+fun Foldable.Companion.setK(): Foldable<ForSetK> = object : Foldable<ForSetK> {
```

Comment:
These should not get new implementations since `Foldable`, `SemigroupK`, `MonoidK`, `Semigroupal`, `Monoidal`, `EqK` will be removed.